### PR TITLE
handle error thrown by status task

### LINKED ISSUE #527: Improve no deployment error message in status
If you run the status command with a non-existant deployment id you get:

image

At the least the stack trace should be removed. Maybe a pattern to follow is HHs:

image

### PR DESCRIPTION
**Pull request by zoeyTM**
fixes #527 

Screen Shot 2023-10-10 at 5 09 59 PM

### CODE REVIEW

File: packages/hardhat-plugin/src/index.ts [hunk 1]
```diff
@@ -247,7 +248,15 @@ ignitionScope
       id
     );
 
-    const statusResult = await status(deploymentDir);
+    let statusResult: StatusResult;
+    try {
+      statusResult = await status(deploymentDir);
+    } catch (e) {
+      throw new NomicLabsHardhatPluginError(
```
Comment (alcuadrado) [inline packages/hardhat-plugin/src/index.ts:255]: I think it'd be better to selectively rethrow, something like this:


```
catch (e) {
  if (e instanceOf IgnitionError && shouldBeHardhatPluginError(e)) {
    throw new ...
  }
  
  throw e;
}
```

```
function shouldBeHardhatPluginError(error:IgnitionError):boolean {
   /// white list of errors by their code
}
```

Comment (alcuadrado) [reply packages/hardhat-plugin/src/index.ts:255]: that way we can use that logic on all the tasks, and make sure we are handling all the errors we care about

Comment (zoeyTM) [reply packages/hardhat-plugin/src/index.ts]: done!

**Review by kanej — APPROVED** [review]
I am going to merge this.

I think we should reuse it across our other tasks, but lets do that as a separate PR.

**Pull request merged**